src/ol/dom.js

```diff
@@ -19,22 +19,24 @@ export function createCanvasContext2D(
   opt_canvasPool,
   opt_Context2DSettings
 ) {
-  const canvas =
+  const canvas = /** @type {HTMLCanvasElement} */ (
```

Comment:
An `OffscreenCanvas` is not a `HTMLCanvasElement`. So I think you'll need to enter two different paths for `OffscreenCanvas` and `HTMLCanvasElement`.

Reply:
`OffscreenCanvas` does not have a style object which is is the reason why the `if (canvas.style) {` test is needed to prevent an error.  Apart from that the properties and methods are the same as the end result was already typecast as `CanvasRenderingContext2D`.  It is one typecast instead of three.

Reply:
I was envisioning a change without any additional type cast. Like this:
```diff
--- a/src/ol/dom.js
+++ b/src/ol/dom.js
@@ -19,12 +19,16 @@ export function createCanvasContext2D(
   opt_canvasPool,
   opt_Context2DSettings
 ) {
-  const canvas =
-    opt_canvasPool && opt_canvasPool.length
-      ? opt_canvasPool.shift()
-      : WORKER_OFFSCREEN_CANVAS
-      ? new OffscreenCanvas(opt_width || 300, opt_height || 300)
-      : document.createElement('canvas');
+  /** @type {HTMLCanvasElement|OffscreenCanvas} */
+  let canvas;
+  if (opt_canvasPool && opt_canvasPool.length) {
+    canvas = opt_canvasPool.shift();
+  } else if (WORKER_OFFSCREEN_CANVAS) {
+    canvas = new OffscreenCanvas(opt_width || 300, opt_height || 300);
+  } else {
+    canvas = document.createElement('canvas');
+    canvas.style.all = 'initial';
+  }
   if (opt_width) {
     canvas.width = opt_width;
   }
```